enterprise/internal/batches/sources/gitlab.go

```diff
@@ -412,6 +415,13 @@ func (s *GitLabSource) UndraftChangeset(ctx context.Context, c *Changeset) error
 	return s.UpdateChangeset(ctx, c)
 }
 
+// UpdateDraftChangeset updates a Changeset on the source, being careful to
+// retain its draft status.
+func (s *GitLabSource) UpdateDraftChangeset(ctx context.Context, c *Changeset) error {
+	c.Title = gitlab.SetWIP(c.Title)
+	return s.UpdateChangeset(ctx, c)
```

Comment:
Thought:

Wouldn't it be simpler to let the source decide what to do, instead of adding a new method to the generic Source interface which all need to implement but only matters to GitLab? Seems like it'd be exposing an implementation detail 🤔

```suggestion
    mr, ok := c.Changeset.Metadata.(*gitlab.MergeRequest)
	if !ok {
		return errors.New("Changeset is not a GitLab merge request")
	}
	if mr.WorkInProgress {
	  c.Title = gitlab.SetWIP(c.Title)
	}
	return s.UpdateChangeset(ctx, c)
}
```

Reply:
I _think_ this is even more correct. If the changeset was draft and is closed, a reopen and update call might read "closed" instead of "draft" and secretly undraft the changeset (but haven't verified that)

Reply:
Oh, that's clever. I had tried to come up with a way to make it implicit rather than adding a new call, but I hadn't found something I liked, and I think that's because I forgot it's safe to downcast the metadata once you're in the actual changeset source.

Thanks; I'll play around with that now.

Reply:
Gah, we even _had_ the downcasted MR in the function already and I didn't think of using it for this. Thanks, that was the perfect suggestion.